sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/sql/BeamRecordSize.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Types;
+import java.util.Map;
+
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.coders.VarLongCoder;
+import org.apache.beam.sdk.extensions.sql.BeamRecordSqlType;
+import org.apache.beam.sdk.nexmark.model.KnownSize;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.BeamRecord;
+
+/**
+ * KnownSize implementation to estimate the size of a BeamRecord, similar to Java model.
+ * NexmarkLauncher/Queries infrastructure expects the events to be able to quickly provide
+ * the estimates of their sizes.
+ *
+ * <p>The BeamRecord size is calculated at creation time.
+ *
+ * <p>Field sizes are sizes of Java types described in {@link BeamRecordSqlType}. Except strings,
+ * which are assumed to be taking 1-byte per character plus 1 byte size.
```

Comment:
<!--thread_id:cc_154870923_t; commit:dab81baf15038727942d0f963bd402793a80ec30; resolved:1-->
<!--section:context-quote-->
> **kennknowles** wrote:
> This is fine for an estimate of the benchmark as long as it has no operational significance (yet).

<!--section:body-->
I am not sure how usable or valuable these metrics are at the moment. 

It's the same assumption current java model has. It's based on what the coder does for the kinds of strings we have here.